src/ripple/basics/Buffer.h

```diff
@@ -191,6 +191,26 @@ class Buffer
     {
         return alloc(n);
     }
+
+    auto begin() noexcept
```

Comment:
These new `begin` and `end` functions should all be `const`. Here and below (expecially for the key types). I don't think we want an interface where we write to key types through `begin` and `end`.

Reply:
`begin` and `end` are much more commonly used than `cbegin` and `cend`. I think it's good to provide `cbegin` and `cend`, but we also need to provide `begin const` and `end const`.

Reply:
Sounds good. Thanks for implementing this. I simply cherry-picked it.